### PR TITLE
docs(readme): create options, add more headings

### PR DESCRIPTION
**Pull request by julie-ng**
As a developer, I want complete samples incl. minimum client configuration options, which would include the redis client password. 

I had to manually open the [ioredis](https://github.com/luin/ioredis) documentation to check if it expected a `password` or `secret` configuration option. Probably the former. But including an example here saves first time users the trouble of looking it up - thanks 🙂 🙏

### Original 

In the original, only the hostname was provided. 

```js
fastify.register(require('fastify-redis'), { host: '127.0.0.1' })
// or
fastify.register(require('fastify-redis'), { url: 'redis://127.0.0.1', /* other redis options */ })
```

Btw - why was it `host` in the first example and then `url`?

### Suggestion

Changes the 2nd example to include some common options, esp. `password`

```js
fastify.register(require('fastify-redis'), { 
  host: '127.0.0.1', 
  password: '***',
  port: 6379, // Redis port
  family: 4   // 4 (IPv4) or 6 (IPv6)
})
```

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,26 +7,42 @@
...
+
+Under the hood [ioredis](https://github.com/luin/ioredis) is used as client, the ``options`` that you pass to `register` will be passed to the Redis client.
 
+```js
 const fastify = require('fastify')()
 
+// minimum options
 fastify.register(require('fastify-redis'), { host: '127.0.0.1' })
-// or
-fastify.register(require('fastify-redis'), { url: 'redis://127.0.0.1', /* other redis options */ })
```
Comment (darkgl0w) [inline README.md:28]: The `url` option is valid too. It should not be removed from the documentation examples.

Can you add it back with a dedicated example ?
And maybe adding a note saying that passing a redis url is another valid way of registering the plugin.

**Review by darkgl0w — CHANGES_REQUESTED** [review]
Thank you for your PR :)

A little nit.
LGTM with the suggested change. ^^

**Review by mcollina — APPROVED** [review]
lgtm

File: README.md (hunk 1 above)
Comment (julie-ng) [reply README.md:28]: Done. Has to reword the comments. Let me know if the wording works for you.

**Review by Eomm — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -7,26 +7,45 @@
...
+  host: '127.0.0.1', 
+  password: '***',
+  port: 6379, // Redis port
+  family: 4   // 4 (IPv4) or 6 (IPv6)
+})
+```
+
+### Accessing the Redis Client
+
+Once you're registered your plugin, you can access the Redis client via `fastify.redis`.
```
Comment (Fdawgs) [inline README.md:44]:
```suggestion
Once you have registered your plugin, you can access the Redis client via `fastify.redis`. 
```

File: README.md [hunk 3]
```diff
@@ -7,26 +7,45 @@
...
+Under the hood [ioredis](https://github.com/luin/ioredis) is used as client, the ``options`` that you pass to `register` will be passed to the Redis client.
+
+```js
 const fastify = require('fastify')()
 
+// create by specifying host
 fastify.register(require('fastify-redis'), { host: '127.0.0.1' })
-// or
+
+// OR or by specifying Redis URL
```
Comment (Fdawgs) [inline README.md:30]:
```suggestion
// OR by specifying Redis URL
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by darkgl0w — APPROVED** [review]
LGTM with @Fdawgs suggested changes. ^^

**Comment by julie-ng** [comment]
@Fdawgs okay your changes have been applied :)

File: README.md [hunk 4]
```diff
@@ -7,26 +7,45 @@
...
+})
+```
+
+### Accessing the Redis Client
+
+Once you have registered your plugin, you can access the Redis client via `fastify.redis`. 
+
+The client is automatically closed when the fastify instance is closed.
+
+```js
```
Comment (darkgl0w) [inline README.md:48]:
```suggestion
```js
'use strict'

const Fastify = require('fastify')
const fastifyRedis = require('fastify-redis')

const fastify = Fastify({ logger: true })

fastify.register(fastifyRedis, { 
  host: '127.0.0.1', 
  password: 'your strong password here',
  port: 6379, // Redis port
  family: 4   // 4 (IPv4) or 6 (IPv6)
})

```

Missed that this example was cut.

Comment (julie-ng) [reply README.md:48]: That will teach me to blinding click "commit suggestion" 🙈

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**